labgrid/driver/resetdriver.py

```diff
@@ -24,3 +27,61 @@ def reset(self):
         self.output.set(True)
         time.sleep(self.delay)
         self.output.set(False)
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class BCUResetDriver(Driver, ResetProtocol, BootCfgProtocol):
+    """BCUResetDriver - Driver using https://github.com/NXPmicro/bcu and board
+    serial port as USBResetPort to reset the target into different states"""
+
+    bindings = {
+       "port": {USBResetPort, NetworkUSBResetPort}, 
+    }
+
+    delay = attr.ib(default=1000, validator=attr.validators.instance_of(int))
+    emmc_cfg = attr.ib(default="emmc", validator=attr.validators.instance_of(str))
+    sd_cfg = attr.ib(default="sd", validator=attr.validators.instance_of(str))
+    usb_cfg = attr.ib(default="usb", validator=attr.validators.instance_of(str))
+    qspi_cfg = attr.ib(default="", validator=attr.validators.instance_of(str))
```

Comment:
Mostly they are the same , but there are cases when they are different- usually qspi_cfg is different from board to board, and for the rest it might be the case where you need m_usb or m_emmc ... like from M core and not A core. Also you may have different emmc config for the same board and you need to select the right one: like for 8ULP
example config:

> BCUResetDriver:
>       emmc_cfg: 'emmc_s'

How it looks in bcu:

> sd@sd-01:~$ bcu lsbootmode -board=imx8ulpevk
> version bcu_1.1.52-0-g17ab144
> board model is imx8ulpevk
> 
> available boot mode:
> 
>         fuse
>         usb
>         emmc_s
>         emmc_nor_lp
>         emmc_nor
>         nand_nor
>         nor_s
>         nor_nor_lp
>         nor_nor
> done
> 
> 
>

Reply:
Hmm, so you have 'emmc' as an alias which is used by the strategy and then maps to different boot modes on different places/boards?

Reply:
in strategy I have a declaration like this (the controller part is related to the second driver that ca switch between boot modes , that one also can controll power):
``` 
   def __attrs_post_init__(self):
        super().__attrs_post_init__()
        if "controller" in self.target.env.config.get_features():
            self.switchcfg = self.power
        if "bcu" in self.target.env.config.get_features():
            self.switchcfg = self.target.get_driver(BCUResetDriver)
```
and then I have a state like this (state cycle reffer to power cycle of the board):
``` 
       elif status == Status.emmc:
            self.transition(Status.cycle)
            try:
               self.switchcfg.emmc()
            except:
               print("Board doesn't support remote switching or this state is not configured in board.yaml")
```

This means that for a board that has the configuration like this:
```
    drivers:
      BCUResetDriver:
        qspi_cfg: 'spi'
```
it will call bcu with bootmode emmc (the default config): ` bcu reset emmc -board=.. -id=..`
and for a board that has this configuration:
 ```
    drivers:
      BCUResetDriver:
        emmc_cfg: 'emmc_s'
```
it will call bcu like this (using the config from yaml): ` bcu reset emmc_s -board=.. -id=..`